### ISSUE TITLE
Adding Ref to LiveKitRoom for TrackToggles

### ISSUE THREAD

**Issue by jjensen-ai** [livekit/components-js#373]
### Describe the problem

Currently we're trying to produce a discord like application using NextJS. Ideally what we would like to do is have the TrackToggle controls accessible from outside of where the room is. Ideally so you can mute the mic  and switch your camera on and off from a tool bar that isnt in the room itself if you're chatting from a different room. I understand as it is so far if you put the tracktoggles outside of the livekitroom it errors as its outside of context. 

### Describe the proposed solution

Ideally for this idea to work in our heads at the present moment would be that you could put the LivekitRoom in the index of the nextJS application  and have it wrap everything else, but give it a ref tag that we could use in tandem with react useRef, and store that as a context to serve to  the component that will launch the LiveKitRoom from that component, but still have the control of the livekitroom from tracktoggles from other components of the application. 

Something sort of like this

```const liveKitRef = useRef();
<LiveKitRoom ref={liveKitRef}>
  <LiveKitContext.Provider value={{room: liveKitRef}}>
    <RenderDesktopView>
      ...
        <Stream>
          const room = useLiveKitRoom(); // Where useLiveKitRoom serves as a getter from the LiveKitContext to return the liveKitRef

          const token = useToken();

          room.token = token; 
        </Stream>
    </RenderDesktopView>
  </LiveKitContext.Provider>
</LiveKitRoom>```

### Alternatives considered

_No response_

### Importance

would make my life easier

### Additional Information

_No response_

**Comment by lukasIO** [comment]
hi @jjensen-ai, 
I'll give you an example of how this could be implemented today with what the components already offer and would be glad to hear your feedback on it: 

### Storing a reference to room somewhere in your project root/index
```tsx
const [room] = React.useState(new Room());

<LiveKitRoom room={room} .... >
 {/* ... */ }
</LiveKitRoom>
```

Note that `<LiveKitRoom />` already creates a room context internally which you can access every where underneath it with 
```tsx
const room = useRoomContext(); // side note: this is just a reference to the object, there aren't any state updates handled
```

### Re-using LiveKit room outside of <LiveKitRoom/> component
you can also spawn your own room context e.g. when you want the controls to live outside of the room component by providing the room context explicitly to a tsx tree:


```tsx
  const [room] = React.useState(new Room());

  return (
    <div data-lk-theme="default">
      <LiveKitRoom
        room={room}
        video={true}
        audio={true}
        token={token}
        serverUrl={process.env.NEXT_PUBLIC_LK_SERVER_URL}
      >
        <VideoConference />
      </LiveKitRoom>

      <div className="custom-controls">
        <RoomContext.Provider value={room}>
          <ControlBar />
        </RoomContext.Provider>
      </div>
    </div>
  );
```

**Comment by jjensen-ai** [comment]
Hi @lukasIO 

I will definitely keep this in mind for another project so i can experiment more with LiveKit in different ways. 

I did end up coming to a solution on my own in order to get the toggles working anywhere in our application.

What I ended up doing is wrapping all of the children in the index of our application with the LiveKitRoom, therefore everything is technically a child of the livekit room, and I've been able to pass the states of our camera, microphone and screen share from anywhere in the application. 

What I ended up doing was when you join any of our media only channels, the token is being passed as a reference from the room back to the top of the application where the livekit room wrapper is and the token ends up being used that way. 

And then for the toggles im just doing a check to see if its enabled via useLocalParticipant, as we're using our own icons for the toggles. 

Thanks for the response, and I just gotta say LiveKits a blessing to be able to get something working quickly, so thank you very much for that!
